packages/framework/aqueduct/src/data-object-factories/pureDataObjectFactory.ts

```diff
@@ -259,8 +259,25 @@ export class PureDataObjectFactory<
 		runtime: IContainerRuntime,
 		initialState?: I["InitialState"],
 	): Promise<TObj> {
-		const context = runtime.createDetachedRootDataStore([this.type], rootDataStoreId);
-		return this.createInstanceCore(context, initialState);
+		const dataStore = (runtime as ContainerRuntime).createDataStore2([this.type]);
+		const { instance, runtime: dataStoreRuntime } = await createDataObject(
+			this.ctor,
+			dataStore.context,
+			this.sharedObjectRegistry,
+			this.optionalProviders,
+			this.runtimeClass,
+			false, // existing
+			initialState,
+		);
+		await dataStore.attachRuntime(this, dataStoreRuntime);
+		const result = await dataStore.trySetAlias(rootDataStoreId);
```

Comment:
Not breaking this sequence into steps might be problematic for users. I.e. I'd expect that some users want to poke at such data store to initialize it properly before attempting to alias it. For example, someone might want to put content into this object (and ideally this is not done through "config" aka initialState, but through component API surface).

Reply:
Is there an expectation that any users other than Loop will use this api? If so then yeah, I would need to really do some much more complex api changes.